# Practical Exercise (Quiz)

We are going to work on a modified version of the dataframe we used at `Analyse et Manipulation de Données`.

In [1]:
import pandas as pd

melb_housing_df = pd.read_csv("./melb_housing_df.csv")

In [2]:
melb_housing_df

housing_address  housing_room_count housing_type  housing_price  \
0          85 Turner St                   2            h      1480000.0   
1       25 Bloomburg St                   2            h      1035000.0   
2          5 Charles St                   3            h      1465000.0   
3      40 Federation La                   3            h       850000.0   
4           55a Park St                   4            h      1600000.0   
...                 ...                 ...          ...            ...   
13201      12 Strada Cr                   4            h      1245000.0   
13202     77 Merrett Dr                   3            h      1031000.0   
13203       83 Power St                   3            h      1170000.0   
13204      96 Verdon St                   4            h      2500000.0   
13205        6 Agnes St                   4            h      1285000.0   

      housing_selling_method housing_seller_agency housing_date_sold  \
0                          S                Biggin         3/12/2016   
1                          S                Biggin         4/02/2016   
2                         SP                Biggin         4/03/2017   
3                         PI                Biggin         4/03/2017   
4                         VB                Nelson         4/06/2016   
...                      ...                   ...               ...   
13201                      S                 Barry        26/08/2017   
13202                     SP              Williams        26/08/2017   
13203                      S                 Raine        26/08/2017   
13204                     PI               Sweeney        26/08/2017   
13205                     SP               Village        26/08/2017   

       housing_cbd_distance  housing_bedroom_count  housing_bathroom_count  \
0                       2.5                    2.0                     1.0   
1                       2.5                    2.0                     1.0   
2                       2.5                    3.0                     2.0   
3                       2.5                    3.0                     2.0   
4                       2.5                    3.0                     1.0   
...                     ...                    ...                     ...   
13201                  16.7                    4.0                     2.0   
13202                   6.8                    3.0                     2.0   
13203                   6.8                    3.0                     2.0   
13204                   6.8                    4.0                     1.0   
13205                   6.3                    4.0                     1.0   

       ...  housing_building_area  housing_year_built  housing_lattitude  \
0      ...                    NaN                 NaN          -37.79960   
1      ...                   79.0              1900.0          -37.80790   
2      ...                  150.0              1900.0          -37.80930   
3      ...                    NaN                 NaN          -37.79690   
4      ...                  142.0              2014.0          -37.80720   
...    ...                    ...                 ...                ...   
13201  ...                    NaN              1981.0          -37.90562   
13202  ...                  133.0              1995.0          -37.85927   
13203  ...                    NaN              1997.0          -37.85274   
13204  ...                  157.0              1920.0          -37.85908   
13205  ...                  112.0              1920.0          -37.81188   

       housing_longitude  suburb_id  \
0              144.99840          0   
1              144.99340          0   
2              144.99440          0   
3              144.99690          0   
4              144.99410          0   
...                  ...        ...   
13201          145.16761        248   
13202          144.87904        104   
13203          144.88738        104   
13

**Exercises**:

**1)** Remove outliers of the random variable `housing_price` by filtering out
all those instances which are 2.5 standard deviations apart from the mean. That
is, given an instance $x$ and mean $\overline{x}$ and standard deviation
$\sigma$, $x$ is **not** considered **an outlier** iff:

$$|x - \overline{x}| \leq 2.5 \times \sigma $$

Apply the filter by updating the dataframe `melb_housing_df`.

**2)** Using the filtered dataframe `melb_housing_df` of exercise 1) and the
columns `housing_seller_agency` and `housing_price`, obtain the total collected
money of the seller sales. Then, select the sellers with the 10 highest earnings
(Top 10 best sellers).

**3)** Create a `barplot` using `seaborn` that compares on the X ax the agency
of these 10 sellers and their average property price.

**4)** Create the same `barplot` but this time using `plotly`. Remember that a
barplot performs an aggregation that in seaborn is done automatically, but in
Plotly, you'll have to make the aggregation before with pandas. Do not consider
the error bars for this plot.